Bottom: 5d9447b8e71d7fa0e67bb489ca9da543b7704542
Top:    570a1ef67fccedfb4c678acf1e988a99e1ec79e8
Author: Jon Smirl <jonsmirl@terra>
Date:   2012-09-09 14:57:47 -0400

From b485de4ab2a5e7ff22f4e7b5a19c470a2c9f4db7 Mon Sep 17 00:00:00 2001

Subject: Make sure mmc stop command is fully initialized before issuing the
 parent command


---

diff --git a/drivers/mmc/host/lpc31xx_mmc.c b/drivers/mmc/host/lpc31xx_mmc.c
index f04d4c7..31b8f78 100644
--- a/drivers/mmc/host/lpc31xx_mmc.c
+++ b/drivers/mmc/host/lpc31xx_mmc.c
@@ -611,11 +611,10 @@ static void lpc313x_mci_start_request(struct lpc313x_mci *host,
 	if (data) //we may need to move this code to mci_start_command
 		lpc313x_mci_submit_data(host, data);
 
-	lpc313x_mci_start_command(host, cmd, cmdflags);
-
 	if (mrq->stop) 
 		host->stop_cmdr = lpc313x_mci_prepare_command(slot->mmc, mrq->stop);
-	
+
+	lpc313x_mci_start_command(host, cmd, cmdflags);
 }
